core/src/main/java/com/sequenceiq/cloudbreak/fluent/FluentConfigService.java

```diff
@@ -1,103 +1,160 @@
-                    )
-                    .build();
-        } else {
-            return new FluentConfigView.Builder()
-                    .withEnabled(false)
-                    .build();
+    private void fillS3Configs(FluentConfigView.Builder builder, String storageLocation,
+            String clusterType, String clusterName, String logFolderName) {
+        BucketFolderPrefixPair bucketFolderPrefixPair = generateBucketFolderPrefixPair(storageLocation);
+        logFolderName = resolveLogFolder(logFolderName, bucketFolderPrefixPair.folderPrefix, clusterType, clusterName);
+
+        builder.withProviderPrefix(S3_PROVIDER_PREFIX)
+                .withS3LogArchiveBucketName(bucketFolderPrefixPair.bucket)
+                .withLogFolderName(logFolderName);
+    }
+
+    // TODO: add support for Azure MSI
+    private void fillWasbConfigs(FluentConfigView.Builder builder, String storageLocation,
+            WasbCloudStorageV1Parameters wasbParams, String clusterType, String clusterName, String logFolderName) {
+        AzureBlobContainerParts azureBlobContainerParts = generateAzureBlobContainerParts(storageLocation);
+        logFolderName = resolveLogFolder(logFolderName, azureBlobContainerParts.folderPrefix, clusterType, clusterName);
+        String storageAccount = StringUtils.isNotEmpty(azureBlobContainerParts.account)
+                ? azureBlobContainerParts.account : wasbParams.getAccountName();
+
+        builder.withProviderPrefix(WASB_PROVIDER_PREFIX)
+                .withAzureStorageAccessKey(wasbParams.getAccountKey())
+                .withAzureContainer(azureBlobContainerParts.storageContainer)
+                .withAzureStorageAccount(storageAccount)
+                .withLogFolderName(logFolderName);
+    }
+
+    private String resolveLogFolder(String logFolderName, String folderPrefix, String clusterType, String clusterName) {
+        if (StringUtils.isNotEmpty(folderPrefix)) {
+            logFolderName = Paths.get(folderPrefix, clusterType, clusterName).toString();
         }
+        return logFolderName;
     }
 
     private BucketFolderPrefixPair generateBucketFolderPrefixPair(String location) {
         if (StringUtils.isNotEmpty(location)) {
-            Path path = getPathWithoutSchemePrefixes(location, S3_SCHEME_PREFIXES);
-            String[] splitted = path.toString().split("/", 2);
+            String locationWithoutScheme = getLocationWithoutSchemePrefixes(location, S3_SCHEME_PREFIXES);
+            String[] splitted = locationWithoutScheme.split("/", 2);
+            String folderPrefix = splitted.length < 2 ? null :  splitted[1];
+            return new BucketFolderPrefixPair(splitted[0], folderPrefix);
+        }
+        throw new CloudbreakServiceException("Storage location parameter is missing for S3");
+    }
+
+    private AzureBlobContainerParts generateAzureBlobContainerParts(String location) {
```

Comment:
Extract this method and `generateBucketFolderPrefixPair` into a class called `CloudWhateverGenerator` and unit test these methods.
Replace whatever with the name of what's common in `BucketFolderPrefixPair` and `AzureBlobContainerParts`. E.g. to `CloudStorageDataHolderGenerator`, or something.